### PR TITLE
 feature/association-user-and-financial-transaction-category

### PR DESCRIPTION
**Pull request by SzanieckiFranciszek**
#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2610; New code is covered with tests and coverage requirements are met.
&#x2610; Build is not failing.
&#x2610; All tests are passing.
&#x2610; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2610; All issue requirements are implemented.
&#x2610; At least two reviewers are assigned.
&#x2610; The Pull request is linked to the related issue.
&#x2610; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2610; Correct unit test name - suffix 'Test' in the end.
&#x2610; Correct integration test name - suffix 'IT' in the end.
&#x2610; Positive and negative test paths have been implemented.

### CODE REVIEW

File: src/main/resources/db/changelog/0011-add-key-constraint-user-and-financial-transaction-category.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+databaseChangeLog:
+  - changeSet:
+      id: "add foreign key to financial_transaction_categories"
+      author: SzanieckiFranciszek
```
Comment (Iwona007) [inline src/main/resources/db/changelog/0011-add-key-constraint-user-and-financial-transaction-category.yaml:4]: Please delete "-" mark

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java [hunk 2]
```diff
@@ -74,4 +80,11 @@ public FinancialTransactionCategory(String name, FinancialTransactionType type)
...
     }
 
+    public void addUser(User user){
+        this.setUser(user);
+    }
+
+    public void removeUser(){
+        this.setUser(null);
+    }
 }
```
Comment (Iwona007) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:90]: These line 83-89 are necessary. Helper methods: add and remove should be on "to many" table. In this case in financialTransactionCategory

File: src/main/resources/db/changelog/0003-create-table-financial_transaction_categories.yaml [hunk 3]
```diff
@@ -26,5 +26,17 @@ databaseChangeLog:
...
+      id: "add column user_id to financial transaction categories"
+      author: SzanieckiFranciszek
+      changes:
+        - addColumn:
+            tableName: financial_transaction_categories
+            columns:
+              - column:
+                  name: user_id
+                  type: bigint
                   constraints:
```
Comment (Iwona007) [inline src/main/resources/db/changelog/0003-create-table-financial_transaction_categories.yaml:41]: Why didn't you implement this as a new databaseChangeLog?

**Review by Iwona007 — COMMENTED** [review]
Changes are required

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes

File: src/main/resources/db/changelog/0011-add-key-constraint-user-and-financial-transaction-category.yaml (hunk 1 above)
Comment (SzanieckiFranciszek) [reply src/main/resources/db/changelog/0011-add-key-constraint-user-and-financial-transaction-category.yaml:4]: Hi, I removed "_" from line 3, this is what you means? In line 2, mark "-" is it good practise followed in other .yaml file.

File: src/main/resources/db/changelog/0003-create-table-financial_transaction_categories.yaml (hunk 3 above)
Comment (SzanieckiFranciszek) [reply src/main/resources/db/changelog/0003-create-table-financial_transaction_categories.yaml:41]: Corrected. I Created new file: 0012-add-new-column-to-financial-transaction-category.yaml.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java (hunk 2 above)
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:90]: Done, I removed redundant methods from User class.

**Review by Iwona007 — DISMISSED** [review]
All previous changes  has been implemented. Approve

**Review by mirekgab — CHANGES_REQUESTED** [review]
The class `FinancialTransactionCategoryServiceImpl`, the method `createFinancialTransactionCategory`. Set the user in the entityToSave as follow:
```java
    @Override
    public FinancialTransactionCategoryDTO createFinancialTransactionCategory(@Valid
            FinancialTransactionCategoryCreateDTO dto) {

        FinancialTransactionCategory entityToSave = new FinancialTransactionCategory(dto.name(), dto.type());
        
        //TODO should be changed to the userId received from the controller
        User user = userRepository.getReferenceById(1L);
        entityToSave.setUser(user);
        
        FinancialTransactionCategory savedEntity = financialTransactionCategoryRepository.save(entityToSave);

        return financialTransactionCategoryModelMapper
                .mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(savedEntity);
    }
```

In the class `FinancialTransactionCategoryServiceImplTest` add 
```java
@Mock
private UserRepository userRepository;
```

**Review by dawciobiel — APPROVED** [review]
Looks good to me

**Pull request closed without merging**